### PR TITLE
Upgrade to akka 2.4.0

### PR DESCRIPTION
**Pull request by johanandren**
2.4.0-RC1 for now, so don't merge yet.

### CODE REVIEW

File: src/main/java/HelloAkkaJava.java [hunk 1]
```diff
@@ -40,35 +41,40 @@ else if (message instanceof Greet)
...
 
-        // Change the greeting and ask for it again
-        greeter.tell(new WhoToGreet("typesafe"), ActorRef.noSender());
-        inbox.send(greeter, new Greet());
-        Greeting greeting2 = (Greeting) inbox.receive(Duration.create(5, TimeUnit.SECONDS));
-        System.out.println("Greeting: " + greeting2.message);
+            // Change the greeting and ask for it again
+            greeter.tell(new WhoToGreet("typesafe"), ActorRef.noSender());
+            inbox.send(greeter, new Greet());
+            Greeting greeting2 = (Greeting) inbox.receive(Duration.create(5, TimeUnit.SECONDS));
```
Comment (ktoso) [inline src/main/java/HelloAkkaJava.java]: `final` to fit style of the rest of the code?

Comment (johanandren) [inline src/main/java/HelloAkkaJava.java]: Good point! I'll fix that.

File: src/main/java/HelloAkkaJava.java [hunk 2]
```diff
@@ -40,35 +41,40 @@ else if (message instanceof Greet)
...
+            final Greeting greeting2 = (Greeting) inbox.receive(Duration.create(5, TimeUnit.SECONDS));
+            System.out.println("Greeting: " + greeting2.message);
 
-        // after zero seconds, send a Greet message every second to the greeter with a sender of the GreetPrinter
-        ActorRef greetPrinter = system.actorOf(Props.create(GreetPrinter.class));
-        system.scheduler().schedule(Duration.Zero(), Duration.create(1, TimeUnit.SECONDS), greeter, new Greet(), system.dispatcher(), greetPrinter);
+            // after zero seconds, send a Greet message every second to the greeter with a sender of the GreetPrinter
+            final ActorRef greetPrinter = system.actorOf(Props.create(GreetPrinter.class));
+            system.scheduler().schedule(Duration.Zero(), Duration.create(1, TimeUnit.SECONDS), greeter, new Greet(), system.dispatcher(), greetPrinter);
+        } catch (TimeoutException ex) {
```
Comment (patriknw) [inline src/main/java/HelloAkkaJava.java]: was this added by a change in `inbox` ?
Could you add a section in the migration guide?

**Comment by patriknw** [comment]
LGTM

File: src/main/java/HelloAkkaJava.java (hunk 2 above)
Comment (johanandren) [inline src/main/java/HelloAkkaJava.java]: Done: #18462

**Comment by patriknw** [comment]
https://github.com/typesafehub/activator-hello-akka/pull/35

**Pull request closed without merging**